File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    answer = 0
    deq = deque(enumerate(priorities))
    while deq:
        lo, pri = deq.popleft()
        if any(pri < priority for lo, priority in deq):
            deq.append((lo, pri))
        else:   
            answer += 1
            if lo == location:
                return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<int[]> que = new ArrayDeque<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        int idx = 0;
        for (int p : priorities) {
            que.add(new int[]{p, idx++});
            pq.add(p);
        }
        
        while (true) {
            int[] cur = que.poll();
            if (cur[0] < pq.peek()) {
                que.add(cur);
            } else {
                pq.poll();
                answer++;
                if (cur[1] == location) {
                    break;
                }
            }
        }
        return answer;
    }
}